refactor:

BorodaPT opened the pull request:
add JPA

BorodaPT committed:
refactor:
add JPA

db-exp commented on src/main/java/ru/practicum/shareit/booking/BookingRepository.java:
```diff
@@ -0,0 +1,118 @@
+import java.util.Optional;
+
+
+public interface BookingRepository extends JpaRepository<Booking, Long> {
```
не хватает аннотации Repository

db-exp commented on src/main/java/ru/practicum/shareit/booking/BookingService.java:
```diff
@@ -0,0 +1,23 @@
+@Transactional(readOnly = true)
+public interface BookingService {
+
+    BookingDtoWithItemUser saveNewBooking(BookingDto bookingDto, long userId);
```
вот тут Transactional(readOnly = true) не совсем корректно - нужен просто Transactional

db-exp commented on src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java:
```diff
@@ -0,0 +1,133 @@
+
+    private final BookingRepository bookingRepository;
+
+    private final ItemRepository itemRepository;
```
напомню - лучше "чужие" сервисы, а не чужие репозитории

db-exp commented on src/main/java/ru/practicum/shareit/item/CommentRepository.java:
```diff
@@ -0,0 +1,16 @@
+import java.util.List;
+
+
+public interface CommentRepository extends JpaRepository<Comment, Long> {
```
не хватает аннотации Repository

db-exp commented on src/main/java/ru/practicum/shareit/item/ItemRepository.java:
```diff
@@ -1,19 +1,29 @@
+import java.util.Optional;
 
-    Item create(Item item);
+public interface ItemRepository extends JpaRepository<Item, Long> {
```
не хватает аннотации Repository

db-exp commented on src/main/java/ru/practicum/shareit/item/ItemRepositoryInMemory.java:
```diff
@@ -14,7 +12,7 @@
 
 @Component("itemRepositoryInMemory")
-public class ItemRepositoryInMemory implements ItemRepository {
+public class ItemRepositoryInMemory  {
```
InMemory можно уже удалять

db-exp commented on src/main/java/ru/practicum/shareit/item/dto/CommentDTO.java:
```diff
@@ -0,0 +1,25 @@
+public class CommentDTO {
+
+    private Long id;
+    @NotEmpty
```
я бы не смешивал NotEmpty и NotBlank, а лучше оставить одно и добавить Size

db-exp commented on src/main/java/ru/practicum/shareit/user/UserRepository.java:
```diff
@@ -1,18 +1,8 @@
 import ru.practicum.shareit.user.model.User;
 
-import java.util.List;
+public interface UserRepository extends JpaRepository<User, Long> {
```
не хватает аннотации repository

db-exp commented on src/main/java/ru/practicum/shareit/user/UserRepositoryInMemory.java:
```diff
@@ -11,7 +11,7 @@
 
 @Component("userRepositoryInMemory")
-public class UserRepositoryInMemory implements UserRepository {
+public class UserRepositoryInMemory  {
```
можно удалить InMemory

db-exp commented on src/main/resources/schema.sql:
предлагаю отформатировать - очень трудно читать/править

db-exp commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,16 @@
+DROP TABLE IF EXISTS comments CASCADE;
+DROP TABLE IF EXISTS requests CASCADE;
+
+CREATE TABLE IF NOT EXISTS users ( id BIGINT GENERATED BY DEFAULT AS IDENTITY NOT NULL, name VARCHAR(200) NOT NULL, email VARCHAR(200) NOT NULL, CONSTRAINT pk_user PRIMARY KEY (id), CONSTRAINT UQ_USER_EMAIL UNIQUE (email));
```
200 - много для email

BorodaPT committed:
Delete ItemRepositoryInMemory.java

BorodaPT committed:
Delete UserRepositoryInMemory.java

BorodaPT committed:
refactor:
внесение изменений согласно замечаниям

BorodaPT committed:
refactor:
внесение изменений согласно замечаниям

BorodaPT committed:
Merge branch 'add-bookings' of https://github.com/BorodaPT/java-shareit into add-bookings

db-exp commented on src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java:
```diff
@@ -0,0 +1,133 @@
+import java.util.List;
+
+@Service
+@Transactional(readOnly = true)
```
у тебя в интерфейсе Transactional, а тут readOnly=true, давай лучше оставим в одном месте, а на конкретном методе овверрайд, например - корневая аннотация Transactional(readOnly = true), а для save - Transactional

BorodaPT committed:
refactor:
Установка @Transactional с найстройкой для соответвуюзих методов чтения и изменения
